src/components/Settings/Rounding.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { Header } from 'semantic-ui-react';
+
+import { setRounding, setRoundingDown } from '../../actions/settings';
+import { valueFromEventTarget } from '../../core/dom';
+import Combo from './Combo';
+
+type RoundingProps = {
+  rounding: string;
+  roundingDown: string;
+  setRoundingProp: (value: any) => void;
+  setRoundingDownProp: (value: any) => void;
+}
+
+class Rounding extends Component<RoundingProps> {
+  constructor(props) {
+    super(props);
+    this.onChangeRoundingDown = this.onChangeRoundingDown.bind(this);
+    this.onChangeRounding = this.onChangeRounding.bind(this);
+    this.onRoundingDownDropDownChange = this.onRoundingDownDropDownChange.bind(this);
+    this.onRoundingDropDownChange = this.onRoundingDropDownChange.bind(this);
+  }
+
+  onRoundingDownDropDownChange: (e: Event, data: any) => void;
+
+  onRoundingDropDownChange: (e: Event, data: any) => void;
+
+  onChangeRounding: (e: Event) => void;
+
+  onChangeRoundingDown: (e: Event) => void;
+
+  onRoundingDropDownChange(e, data) {
+    const { setRoundingProp } = this.props;
+    setRoundingProp(data.value);
+  }
+
+  onRoundingDownDropDownChange(e, data) {
+    const { rounding, setRoundingDownProp } = this.props;
+    const roundingInt = parseInt(data.value, 10);
+    if (roundingInt < parseInt(rounding, 10)) {
+      setRoundingDownProp(data.value);
+    }
+  }
+
+
+  onChangeRounding(e) {
+    const { setRoundingProp } = this.props;
+    const roundValue = valueFromEventTarget(e.target);
+    const roundValueInt = parseInt(roundValue, 10);
+    if (roundValueInt <= 60 && roundValueInt > 0) {
+      setRoundingProp(roundValue);
+    }
+    if (roundValue === '') {
+      setRoundingProp(roundValue);
+    }
+  }
+
+  onChangeRoundingDown(e) {
+    const { rounding, setRoundingDownProp } = this.props;
+    const roundDown = valueFromEventTarget(e.target);
+    const roundDownInt = parseInt(roundDown, 10);
+    if (roundDownInt < parseInt(rounding, 10)
+      && roundDownInt <= 60 && roundDownInt >= 0) {
+      setRoundingDownProp(roundDown);
+    }
+    if (roundDown === '') {
+      setRoundingDownProp(roundDown);
+    }
+  }
+
+  render() {
+    const { rounding, roundingDown } = this.props;
+    return (
+      <Fragment>
+
+        <Header as="h3">
+Rounding UP
+        </Header>
+        <Combo title="The minutes that the end time rounds to" rounding={rounding} onInputChange={this.onChangeRounding} onDropDownChange={this.onRoundingDropDownChange} />
```

Comment:
I think we could be more clear to the user if we change the header to "Rounding start and end times" and then making two inputs.
I really like the idea of a combined input you made, but maybe we can use it in a different way.
Want the user can do is choice whether to round up, down, or automatically in a dropdown and after that select the number of minutes.

For the minutes input I think the approach as we do projects is a good one. Have some pre-defined options (like 5, 10, 15, etc) and if they input something which is not on the list: allow to "add".

![image](https://user-images.githubusercontent.com/970834/44134102-8e4fe030-a063-11e8-8003-f5ccac078540.png)

This will just be a raw value, so the dropdown's value is just the number of minutes to round.

The same can be done for end time.